packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -399,22 +408,26 @@ function WorklistDataGrid(props) {
       id: rowIdsToSelect.length === 1
         ? 'Terra.worklist-data-grid.row-selection-template'
         : 'Terra.worklist-data-grid.row-selection-selected-rows-range',
-    }, { row: rangeInfo.anchorRow, endRow: newEndRowIndex })}`;
+    }, { row: startOfRange, endRow: newEndOfRange })}`;
 
     setAriaLiveMessage(ariaMessage);
     onRowSelect(rowIdsToSelect, rowIdsToUnselect);
-  }, [hasSelectableRows, intl, onEnableRowSelection, onRowSelect, rows]);
+  }, [hasSelectableRows, intl, onEnableRowSelection, onRowSelect, rows, getRowsInRangeToUnselect]);
 
   const selectRow = useCallback((rowId, rowIndex) => {
     const rowsToSelect = [];
     const rowsToUnSelect = [];
-    rowSelection.current.rangeUsingLastIndividuallySelectedRow = { anchorRow: rowIndex };
     const rowLabel = rows[rowIndex - 1].ariaLabel || (rowIndex + 1);
 
     if (!rows[rowIndex - 1].isSelected) {
+      multiSelectRange.current = { start: rowIndex, end: null }; // Establish new starting point for future range.
       setAriaLiveMessage(intl.formatMessage({ id: 'Terra.worklist-data-grid.row-selection-template' }, { row: rowLabel }));
       rowsToSelect.push(rowId);
     } else {
+      if (rowIndex === multiSelectRange.current?.start) {
```

Comment:
Based on your logic, the current property of mutliSelectRange would never be null.  There does not seem to be a need for the question mark.